### PR TITLE
[WIP] Switch to NavigationExperimental

### PR DESCRIPTION
**Pull request by ericvicenti**
This isn't fully finished/tested/polished, but it moves F8 over to a declarative navigation setup!

I'd appreciate some help finishing and cleaning this up:

- Card animation needs polish
- Need to write animation card for modals
- Android specific UI and testing
- Test/convert friend page feature (I didn't have any f8 friends on this app)
- Test edge cases with login/logout
- Convert rating and notices screens.

Addresses #8

### CODE REVIEW

**Comment by ptomasroos** [comment]
Great work @ericvicenti !

File: js/F8Navigator.js [hunk 1]
```diff
@@ -88,81 +89,120 @@ var F8Navigator = React.createClass({
...
+  //         }
+  //       }}
+  //       initialRoute={{}}
+  //       renderScene={this.renderScene}
+  //     />
+  //   );
+  // },
+
   render: function() {
+    console.log('bah', this.props.openModals)
```
Comment (ptomasroos) [inline js/F8Navigator.js]: I think we can remove this as well.

File: js/F8Navigator.js [hunk 2]
```diff
@@ -88,81 +89,120 @@ var F8Navigator = React.createClass({
...
+  renderInnerScene: function(props) {
+    const sceneState = props.scene.navigationState;
+    // if (route.allSessions) {
+    //   return (
+    //     <SessionsCarousel
+    //       {...route}
+    //     />
+    //   );
+    // }
+    console.log('renderign scene', sceneState);
```
Comment (ptomasroos) [inline js/F8Navigator.js]: Either remove or rendering*

File: js/actions/schedule.js [hunk 3]
```diff
@@ -131,7 +131,7 @@ function setSharingEnabled(enabled: boolean): ThunkAction {
 }
 
 function shareSession(session: Session): ThunkAction {
-  return (dispatch, getState) => {
+  return (dispatch, getState) => { debugger;
```
Comment (radev) [inline js/actions/schedule.js]: Leftover debugger statement?

**Comment by mikelambert** [comment]
@frantic , curious why this was closed?

**Comment by ericvicenti** [comment]
The history of the repo was revised, and I haven't had a chance to rebase and commit this
